src/test/scala/chiselTests/BundleSpec.scala

```diff
@@ -129,6 +129,17 @@ class BundleSpec extends ChiselFlatSpec with BundleSpecUtils with Utils {
     }).getMessage should include("aliased fields")
   }
 
+  "Bundles" should "not have bound hardware" in {
+    (the[ChiselException] thrownBy extractCause[ChiselException] {
+      ChiselStage.elaborate { new Module {
+        class MyBundle(val foo: UInt) extends Bundle
+        val in  = IO(Input(new MyBundle(123.U))) // This should error: value passed in instead of type
+        val out = IO(Output(new MyBundle(UInt(8.W))))
+
+        out := in
+      } }
+    }).getMessage should include("got value binding")
+  }
```

Comment:
It's possible (but I'm not certain), that children of unbound aggregates still get their binding set to 
It'd be good to add 1 extra test  for a bound aggregate getting passed
```suggestion
  "Bundles" should "not have bound hardware" in {
    (the[ChiselException] thrownBy extractCause[ChiselException] {
      ChiselStage.elaborate { new Module {
        class MyBundle(val foo: UInt) extends Bundle
        val in  = IO(Input(new MyBundle(123.U))) // This should error: value passed in instead of type
        val out = IO(Output(new MyBundle(UInt(8.W))))

        out := in
      } }
    }).getMessage should include("got value binding")
    (the[ChiselException] thrownBy extractCause[ChiselException] {
      ChiselStage.elaborate { new Module {
        class MyBundle(val foo: UInt) extends Bundle
        val out = IO(Output(Vec(2, UInt(8.W))
        val in  = IO(Input(new MyBundle(out(0))) // This should error: value passed in instead of type

        out := in
      } }
    }).getMessage should include("got value binding")
  }
```